### DISCUSSION TITLE
Dump file size

### DISCUSSION THREAD

**Discussion by tuxslayer** [async-profiler/async-profiler#509] [Q&A] [answered]
Hi all!
I'm getting 1+Gb dumps from `AsyncProfiler.dumpCollapsed(SAMPLES)` trying to figure out is that legitimate or I'm doing something wrong and which settings might reduce dump file size to say 100Mb. 
I'm doing only CPU profiling (`start0("cpu", interval)`).

Any help appreciated :)

**Comment by toktarev** [comment]
Looks like you should use "JFR"

Or increase "interval".

**Reply by tuxslayer** [reply]
Will look into that, thanks!

**Comment by apangin** [comment, marked as answer]
- What `interval` do you set?
 - How long is the profiling session (between start and dump)?
 - What are JVM options?

**Reply by tuxslayer** [reply]
We're not setting a lot of params, we just do `start(10000000)/stop/dumpCollapsed` for now with default params. 
So, sampling interval is 10 ms (10000000 nanos).
The length of the session is 10 min, and we got pretty much calls during that period but I would expect sampled stats to be a bit thinner, thus asking.

JVM options, among interesting:
-Djdk.attach.allowAttachSelf=true -server -Dfile.encoding=UTF-8 -Xlog:gc=debug:file=/... 
-XX:+PreserveFramePointer -XX:-OmitStackTraceInFastThrow -XX:+HeapDumpOnOutOfMemoryError -XX:MaxJavaStackTraceDepth=1000000 -XX:+UseG1GC -XX:ParallelGCThreads=16 -XX:ConcGCThreads=4 -Xss256k -XX:StringTableSize=1000003 -XX:MaxMetaspaceSize=2G -Xmx53G -Xms53G -XX:MaxDirectMemorySize=5G -XX:NativeMemoryTracking=summary"

**Reply by apangin** [reply]
OK, the parameters look legit. Well, 10ms interval is a bit too frequent for a 10 min session - I would make 100ms interval for 10 min, or 10ms interval for 1 min, but I don't think this is a root cause of the problem. Probably, your application stack traces are just very deep, and the frame names are long enough to make the entire output so large.

`collapsed` output format is just very redundant. It came from the time when async-profiler required a third-party script for generating a flame graph, but it's not the case anymore, since async-profiler can produce a flame graph out of the box. Take a look at more compact formats: `flamegraph` or `jfr`.

**Comment by toktarev** [comment]
@tuxslayer https://github.com/jvm-profiling-tools/async-profiler/discussions/499#discussioncomment-1661642
